kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -132,16 +175,33 @@
 
   /**
    * Create a {@link Dataset} for the given dataset or view URI.
+   * {@code create} returns an empty dataset. You can use 
+   * {@link DatasetWriter} to populate your dataset.
    * <p>
-   * URI formats are defined by {@code Dataset} implementations, but must begin
-   * with "dataset:" or "view:".
+   * URIs must begin with {@code dataset:} or {@code view:}. The remainder of
+   * the URI is implementation specific, depending on the dataset scheme.
    *
-   * @param uri a {@code Dataset} or {@code View} URI.
-   * @param type the Java type of the entities in the dataset
-   * @param <E> The type of entities stored in the {@code Dataset}.
-   * @param <V> The type of {@code Dataset} or {@code View} expected.
-   * @return a newly created {@code Dataset} responsible for the given URI.
-   */
+   * @param uri a {@code Dataset} or {@code View} URI
+   * @param type a Java class that represents an entity in the dataset
+   * @param <E> the type used for readers and writers created by this {@code Dataset}
+   * @param <V> the type of {@code Dataset} or {@code View} expected
+   * @return a newly created {@code Dataset} responsible for the given URI
+   * @throws IllegalArgumentException   if {@code URI} or {@code descriptor}
+   *                                    is {@code null}
+   * @throws DatasetExistsException     if a {@code Dataset} for the given URI
+   *                                    already exists
+   * @throws ConcurrentSchemaModificationException
+   *                                    if the {@code Dataset}
+   *                                    schema is updated
+   *                                    concurrently
```

Comment:
I don't think we need `ConcurrentSchemaModificationException` here. It should not be thrown for create because we should throw `DatasetExistsException` instead.